lib/pal/PAL.cpp

```diff
@@ -96,9 +96,18 @@ namespace PAL_NS_BEGIN {
         bool isLoggingInited = false;
 
 #ifdef HAVE_MAT_LOGGING
-        std::recursive_mutex          debugLogMutex;
-        std::string                   debugLogPath;
-        std::unique_ptr<std::fstream> debugLogStream;
+
+        std::recursive_mutex& GetDebugLogMutex()
+        {
+            static std::recursive_mutex debugLogMutex;
+            return debugLogMutex;
+        }
+        
+        std::unique_ptr<std::fstream>& GetDebugLogStream() noexcept
```

Comment:
I tried to be minimally invasive with these changes - only moving globals to statically constructed on demand, otherwise preserving the same. If the code was directly accessing unique_ptr global, the static accessor just mirrors that. Having said that I can change it to shared_ptr if preferred